ui/client/documents/upload/index.js

```diff
@@ -249,21 +246,16 @@ const UploadDocumentForm = withStyles((theme) => ({
       }).then(() => {
         history.push('/documents');
       });
-
-      // if we map files to promises to await all uploads:
-      // return documentsPromise;
-
     });
   };
 
-  /**
-   * TODO: In order to handle deletes- either do backflips around deleted indexes,
-   * not rendering deleted ones, etc, or change the collection/representation
-   * of data from arrays and selected index to an object with key=>properties.
-   **/
   const handleFileDelete = (index) => {
-    // TODO Implement me once we use SortedMap
-    console.log("deleting file index:", index);
+    // always reset back to zero, even if we aren't touching the selectedFile
+    // so we don't end up in a weird state where we're modifying the array around this index
+    setSelectedFileIndex(0);
```

Comment:
I was worried about the difference between 0 and null here. When there's one file remaining and we remove it, the index goes back to 0 (instead of null) and we're pointing to no selected file. At one point, during the initial implementation of document upload,  there weren't checks and this would blow up. But I see there are checks below- so this works!

Maybe `selectedFileIndex` should default to `0` instead of `null` (when initializing that portion of state), to make it cleaner now (and remove any null checks for it), but that's not relevant to this PR so we can address it whenever we'd like.